azure_sdk_core/src/modify_conditions.rs

```diff
@@ -10,11 +10,15 @@ pub enum IfSinceCondition {
 }
 
 impl IfSinceCondition {
-    pub(crate) fn add_header(&self, builder: &mut Builder) {
+    pub(crate) fn add_header(&self, builder: Builder) -> Builder {
```

Comment:
We should make sure to specify the `#[must_use]` directive (we can do this in a separate PR).